internal/overlord/servstate/handlers.go

```diff
@@ -77,17 +78,38 @@ func (m *ServiceManager) doStart(task *state.Task, tomb *tomb.Tomb) error {
 
 	_, previous := m.services[req.Name]
 	if previous {
-		return fmt.Errorf("service %q was previously started", req.Name)
+		// Already started
+		return nil
+	}
+
+	// Pass service description's environment variables to child process
+	env := os.Environ()
+	for k, v := range service.Environment {
+		env = append(env, k+"="+v)
+	}
+
+	// Expand the command string with environment variables.
+	envMap := make(map[string]string)
+	for _, envLine := range env {
+		parts := strings.SplitN(envLine, "=", 2)
+		if len(parts) == 2 {
+			envMap[parts[0]] = parts[1]
+		}
 	}
 
-	args, err := shlex.Split(service.Command)
+	commandString := os.Expand(service.Command, func(k string) string {
```

Comment:
I think this is a good change, but it doesn't look like it's documented (in the `README`) or tested anywhere. Can we add those?

Nit: what about calling this `commandExpanded`?

Reply:
Oh yeah I forgot I had made this change, I'll make sure to add some tests and documentation.